components/sensor/dlms_meter.rst

```diff
@@ -0,0 +1,208 @@
+DLMS_Meter Component
+====================
+
+.. seo::
+    :description: Instructions for setting up DLMS Meter component in ESPHome.
+    :image: dlms_meter.jpg
+
+Component
+-------------
+
+The DLMS_Meter component connects to Smart Meters which use the encrypted DLMS/Cosem protocol over uart. These smartmeters are used for example everywhere in Austria.
+
+**To use this component an mbus to uart adapter is required!**
+
+You have to get the decrytion key from your energy provider / grid operator.
+
+This component is passive, it does not transmit any data to your equipment, the equipment always transmits
+data which this component decodes and updates the configured sensors at the pace the data is received (data is usually sent every 5 seconds by most smartmeters).
+
+.. figure:: images/dlms_meter.jpg
+    :align: center
+    :width: 50.0%
+
+    Smartmeter with mbus adapter board to esp32
+
+.. code-block:: yaml
+
+    # Example configuration entry for generic grid operator
+    uart:
+      rx_pin: GPIO4 # change this to the pin where you connected the mbus adapter board
+      baud_rate: 2400
+      rx_buffer_size: 1024 # Needed to receive the large packets send by the smart meter
+
+    dlms_meter:
+      decryption_key: "36C66639E48A8CA4D6BC8B282A793BBB" # change this to your decryption key!
+
+    sensor:
+      - platform: dlms_meter
+        voltage_l1:
+          name: "Voltage L1"
+        voltage_l2:
+          name: "Voltage L2"
+        voltage_l3:
+          name: "Voltage L3"
+        current_l1:
+          name: "Current L1"
+        current_l2:
+          name: "Current L2"
+        current_l3:
+          name: "Current L3"
+        active_power_plus:
+          name: "Active power taken from grid"
+        active_power_minus:
+          name: "Active power put into grid"
+        active_energy_plus:
+          name: "Active energy taken from grid"
+        active_energy_minus:
+          name: "Active energy put into grid"
+        reactive_energy_plus:
+          name: "Reactive energy taken from grid"
+        reactive_energy_minus:
+          name: "Reactive energy put into grid"
+
+    text_sensor:
+      - platform: dlms_meter
+        timestamp:
+          name: "timestamp"
+
+.. collapse:: Example yamls for special grid operators
+
+    NetzNoe / EVN
+
+    .. code-block:: yaml
+
+        # Sample configuration entry for netznow
+        uart:
+          rx_pin: GPIO4 # change this to the pin where you connected the mbus adapter board
```

Comment:
Use the correct RX pin from ESP32 and give a hint that the user had to check what is the correct pin
See: https://github.com/SimonFischer04/esphome/issues/1#issuecomment-2671008121